lib/crl.ml

```diff
@@ -118,30 +118,34 @@ let crl_number { asn ; _ } =
   | None -> None
   | Some (_, x) -> Some x
 
-let validate { raw ; asn } pub =
+let signature_algorithm { asn ; _ } =
+  Algorithm.to_signature_algorithm asn.signature_algo
+
+let validate { raw ; asn } ?(hash_whitelist = Validation.sha2) pub =
```

Comment:
The default value for `hash_whitelist` is hardcoded in several places. Would it make sense to add a `Validation.default_hash_whitelist` value?

Reply:
thanks for your review. I'm not sure what someone would do with a `Validation.default_hash_whitelist`, would you be so kind to elaborate?

when designing and implementing this feature, I had the following use cases in mind:
- a developer who does not care about this at all -> they'll get sane defaults from x509
- a developer who has requirements to disallow certain hash algorithms (or only allow certain ones) -- I'd expect them to hardcode them then

when the set of supported hash algorithms gets modified (i.e. sha3, whatever), this leads to the former not needing any updates, but the latter (who made a decision about the allowed algorithms already) will have to update their code.

does this make sense? who would use the `default_hash_whitelist` in your opinion?

Reply:
that was more a concern for x509 maintainers: when SHA2 gets deprecated, with the current implementation you have to change the default in many places.

If this default gets named, all these definitions become `?(hash_whitelist = Validation.default_hash_whitelist)`, and you won't have to hunt for all uses of `sha2` in the code base: you will just have to update the value of `default_hash_whitelist` so that it aliases the new set of safe algorithms.